server/types.py

```diff
@@ -1,5 +1,23 @@
-from typing import NamedTuple
+from typing import NamedTuple, Optional
 
-# Represents a peer IP address
-Address = NamedTuple('Address', [('host', str), ('port', int)])
-Address.from_string = lambda s: Address(*(s.split(':')))
+
+class Address(NamedTuple):
+    """A peer IP address"""
+
+    host: str
+    port: int
```

Comment:
When i see this, i was thinking, when we will be on 3.7 we might use here dataclasses. https://docs.python.org/3/library/dataclasses.html
Do you have any experience with them? Are they better/worse than named tuples?

Reply:
No I haven't used them before.